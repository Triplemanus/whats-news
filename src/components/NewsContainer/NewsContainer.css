.news-container {
  padding: 20px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
}

.parent {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: 1fr;
  grid-column-gap: 7px;
  grid-row-gap: 0px;
  }