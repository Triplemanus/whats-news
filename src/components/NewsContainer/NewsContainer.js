import React from 'react';
import NewsArticle from '../NewsArticle/NewsArticle'
import './NewsContainer.css'

const NewsContainer = ({news, deleteArticle }) => {
  return (
    news.map(article => {
      return (
      <div className='news-container' >
        <NewsArticle 
          id={article.id}
          key={`${Date.now()}-${article.headline}`}
          image={article.img} 
          headline={article.headline}
          description={article.description}
          url={article.url}
          deleteArticle={deleteArticle}
        />
      </div>
      )
    })
  )
};

export default NewsContainer;