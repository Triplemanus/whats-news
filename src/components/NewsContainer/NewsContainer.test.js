import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  let wrapper;
  const mockDeleteArticle = jest.fn();

  beforeEach(() => {
      wrapper = shallow(<NewsContainer 
      id={1}
      image="http://myimage.com"
      headline="Trump resigns!"
      description="Hell freezes over. Kaiser Drumpf takes his ball and goes home."
      url="http://my.article.com"
      deleteArticle={mockDeleteArticle}
      />);
  });

  it('should match the snapshot', () => {
 
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a news article', () => {

  })
})