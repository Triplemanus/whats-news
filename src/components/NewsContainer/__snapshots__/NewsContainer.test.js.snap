// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `undefined`;

exports[`NewsContainer should match the snapshot 1`] = `undefined`;
